zilliqa/src/uccb/client.rs

```diff
@@ -0,0 +1,62 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+use ethers::{
+    middleware::{MiddlewareBuilder, NonceManagerMiddleware, SignerMiddleware},
+    providers::{Middleware, Provider, Ws},
+    signers::{LocalWallet, Signer},
+    types::{Address, U256},
+};
+
+use crate::uccb::cfg::ChainConfig;
+
+pub type Client = NonceManagerMiddleware<SignerMiddleware<Provider<Ws>, LocalWallet>>;
+
+#[derive(Debug, Clone)]
+pub struct ChainClient {
+    pub rpc_url: String,
+    pub client: Arc<Client>,
+    pub validator_manager_address: Address,
+    // pub chain_gateway_address: Address,
+    pub chain_id: U256,
+    pub wallet: LocalWallet,
+    pub chain_gateway_block_deployed: u64,
+    pub block_instant_finality: bool,
+    pub legacy_gas_estimation: bool,
+}
+
+impl ChainClient {
+    pub async fn new(
+        config: &ChainConfig,
+        validator_manager_address: Address,
+        // chain_gateway_address: Address,
+        wallet: LocalWallet,
+    ) -> Result<Self> {
+        // let provider = Provider::<Http>::try_from(config.rpc_url.as_str())?;
+        let ws = Ws::connect(&config.rpc_url).await?;
+        let provider = ethers::providers::Provider::<Ws>::new(ws);
+        let chain_id = provider.get_chainid().await?;
+
+        let client: Arc<Client> = Arc::new(
+            provider
+                .with_signer(wallet.clone().with_chain_id(chain_id.as_u64()))
+                .nonce_manager(wallet.address()),
+        );
+
+        // TODO: get the validator_manager_address from chain_gateway itself
```

Comment:
Do you plan to address the `TODO`s and commented out bits of code in this PR?

Reply:
Not sure why Bruno added this comment and why it's really important.. but this definitely won't be addressed in this PR.